Server/src/services/dataProcess/JsonFileHandler.ts

```diff
@@ -0,0 +1,200 @@
+import { AbstractFileHandler, FileHandlerFactory } from './IDataProcessService';
+import { DataUploadModel } from '../../models/DataUploadModel'
+import { Authors } from '../../models/entities/Authors';
+import { IMaterials } from '../../models/interfaces/MaterialsInterface';
+import { IAuthors } from '../../models/interfaces/AuthorsInterface';
+import { BadRequest } from '@tsed/exceptions';
+import { validationSchema } from '../helpers/validationSchema';
+import { IJsonDatasetModel } from '../../genericInterfaces/DataProcessInterfaces'
+const fileSystem = require('fs');
+
+
+export class JsonFileFactory extends FileHandlerFactory {
+
+    getFileHandler(filePath: string): AbstractFileHandler {
+        return new JsonFileHandler(filePath)
+    }
+}
+
+export class JsonFileHandler extends AbstractFileHandler {
+
+    async parseFile(): Promise<IJsonDatasetModel> {
+        try {
+            this.parsedFileData = await JSON.parse(fileSystem.readFileSync(this.filePath))
+            return this.parsedFileData
+        } catch (err) {
+            throw new BadRequest("Cannot parse your file. Something is wrong with it");
+        }
+    }
+
+    async validateExtractedData(jsonData: any): Promise<JSON> {
+        try {
+            await validationSchema.validate(jsonData)
```

Comment:
i thought we agreed on no validation? you still think it should be present?

Reply:
This is after the post. This code only gets executed on 'upload'